lib/model/autocomplete.dart

```diff
@@ -684,19 +728,26 @@ abstract class ComposeAutocompleteQuery extends AutocompleteQuery {
 
 /// A @-mention autocomplete query, used by [MentionAutocompleteView].
 class MentionAutocompleteQuery extends ComposeAutocompleteQuery {
-  MentionAutocompleteQuery(super.raw, {this.silent = false});
+  MentionAutocompleteQuery(super.raw, {this.silent = false, required this.localizations});
 
   /// Whether the user wants a silent mention (@_query, vs. @query).
   final bool silent;
 
+  final ZulipLocalizations localizations;
+
   @override
   MentionAutocompleteView initViewModel(PerAccountStore store, Narrow narrow) {
     return MentionAutocompleteView.init(store: store, narrow: narrow, query: this);
   }
 
+  bool testWildcardOption(WildcardMentionOption wildcardOption) {
+    // TODO(#237): match insensitively to diacritics
+    return wildcardOption.localizedCanonicalString(localizations)
+      .contains(_lowercase);
+  }
```

Comment:
This should also check for a match in `wildcardOption.canonicalString`, like in zulip-mobile; not only `localizedCanonicalString`. I had this in mind with https://github.com/zulip/zulip-flutter/pull/889#discussion_r1896021336 but didn't make it super explicit. 🙂 Because `canonicalString` is the string that'll actually be inserted into the message.